src/main/java/com/hedera/hashgraph/sdk/HederaRecordStatusException.java

```diff
@@ -0,0 +1,40 @@
+package com.hedera.hashgraph.sdk;
+
+import com.hedera.hashgraph.proto.ResponseCodeEnum;
+import com.hedera.hashgraph.proto.TransactionGetRecordResponse;
+
+/**
+ * A {@link HederaStatusException}, thrown on error status by {@link TransactionId#getRecord(Client)}.
+ *
+ * The record is included which could contain useful context for the error, such as errors
+ * returned by {@link com.hedera.hashgraph.sdk.contract.ContractExecuteTransaction}.
+ */
+public class HederaRecordStatusException extends HederaStatusException {
+    /**
+     * The ID of the transaction that failed, in case that context is no longer available
+     * (e.g. the exception was bubbled up).
+     */
+    public final TransactionId transactionId;
+
+    /**
+     * The record that was fetched; it may still contain useful context (such as contract execution
+     * logs).
+     */
+    public final TransactionRecord record;
+
+    HederaRecordStatusException(ResponseCodeEnum responseCode, TransactionRecord record) {
+        super(responseCode);
+        this.record = record;
+        this.transactionId = record.transactionId;
+    }
+
```

Comment:
Needs getMessage override so it will display the transactionID and the responsecode.